packages/eslint-plugin/docs/rules/no-explicit-any.mdx

```diff
@@ -145,6 +145,36 @@ interface Garply {
 }
 ```
 
+## Alternatives to `any`
+
+Which type(s) to use instead of `any` depends on the context.
+
```

Comment:
I can't find references of other non-straightforward scenarios to add... are there others we should mention?